src/Sylius/Component/Addressing/Model/Country.php

```diff
@@ -146,6 +146,22 @@ public function hasProvince(ProvinceInterface $province)
     /**
      * {@inheritdoc}
      */
+    public function isEqualTo($data)
+    {
+        if (is_object($data)) {
+            if (!$data instanceof CountryInterface) {
+                return false;
+            }
+
+            return $data->getIsoName() === $this->isoName || $data->getId() === $this->id;
+        }
+
+        return $data === $this->id;
```

Comment:
Totally don't get you...
```php
$id = 1;

$country = new Country()
$country->getId(); // let's say it will return 1

$otherCountry = new Country()
$otherCountry->getId(); // let's say it will return 2

// Checkers call `isEqualTo()`
var_dump($promotion->isEliglible($country, ['country' => $id])); // true
var_dump($promotion->isEliglible($country, ['country' => $country])); // true
var_dump($promotion->isEliglible($country, ['country' => 2])); // false
var_dump($promotion->isEliglible($country, ['country' => $otherCountry])); // false